docs/act.md

```diff
@@ -0,0 +1,195 @@
+# Syntax of the Act language
+
+A specification of a contract written in `act` consists of a
+`constructor` and a set of `behaviours`.
+
+The `constructor` specification defines the structure of the contract
+state, the initial value for the state, and a list of invariants that
+the contract should satisfy.
+
+Each `behaviour` specification determines how a contract method updates
+the state, and its return value, and the conditions that must be
+satisfied in order for it to be applied.
+
+Alternatively, they can be thought of an initial state and a 
+set of state transitions, determining an inductively defined 
+state transition system.
+
+The specifications can be exported to different backends in order to
+prove that the claims that are being made hold with respect to an implementation.
+
+This document is a high level description of the syntax of 
+act specifications.
+For a more formal treatment of the syntax, study the
+definitions of [RefinedAst.hs](https://github.com/ethereum/act/blob/master/src/RefinedAst.hs).
+
+
+## Contructor
+
+A `constructor` specification is indicated by the
+`constructor of <contractName>` header, as in:
+```act
+constructor of A
+```
+
+A constructor section consists of the following fields:
+
+### interface
+
+Specifying the arguments the constructor takes.
+Example:
+```act
+interface constructor(uint _owner)
+```
+
+### iff (optional)
+
+The conditions that must be satisfied in order for the contract to be created.
+These must be necessary and sufficient conditions, in the sense that if any 
+condition isn't met, the contract cannot be created.
+
+Example:
+```act
+iff
+
+CALLER == _owner
+```
+
+
+### creates
+Defines the storage layout of the contract.
+
+Example:
+```act
+
+creates
+
+  uint256 totalSupply := _totalSupply
+  mapping(address => uint) balanceOf :=  [CALLER := _totalSupply]
```

Comment:
would be nice to have a little info on the allowed types here, as well as the initializer list syntax.